### PR TITLE
Phase 4 Slice 2: camera arc + initial entry

### LINKED ISSUE #43: Phase 4 Slice 2: Camera arc + initial entry
## Parent

Part of #41

## What to build

Phase 4 Slice 2. Replace Slice 1's instant snap on selection with a GSAP-style camera arc to the selected country's lat/lon. The arc traces a great-circle direction with an altitude rise at the midpoint ("banking" feel), runs for 900ms with an ease-spring curve, and applies to both selection-change AND initial page load.

The arc is the centerpiece of Phase 4 per the handoff §340 item 5. With Slice 1 already shipping the navigation loop, this slice adds the **motion polish** without changing any data flow.

Motion behavior:

- **Great-circle arc**: SLERP the unit direction from the current camera position to the target country's lat/lon vector.
- **Altitude rise**: radius modulated by `sin(πt)` for a midpoint bump of ~0.15× base radius.
- **900ms duration** (`--dur-globe`), `--ease-spring` easing applied to `t` before passing into the path function.
- **During transition**: `OrbitControls.enabled = false`; subsequent pin clicks are ignored until the arc completes.
- **Sheet timing**: the chart sheet opens to peek in **parallel** with the camera arc (not after).
- **URL**: `?cc=` updates **immediately on click** (not on arrival).
- **Initial page load**: camera starts at a default view (slight pole-tilted) and runs the same arc to land on the picked country in 900ms. Sheet animates in to peek in parallel.

Path generator shape (from PRD; constant tuning left to implementation):

```
cameraArcPath({ from, toLatLon, baseRadius, riseFactor }):
  fromDir = from.normalized()
  toDir = latLonToVec3(toLatLon.lat, toLatLon.lon, 1)

  return (t) =>
    dir = slerp(fromDir, toDir, t)
    r = baseRadius * (1 + riseFactor * sin(π * t))
    return dir * r
```

Ease-spring is applied to `t` before it's passed into the path function — the spatial path stays purely geometric and the easing is timing-only.

New modules:

- `src/lib/camera-arc.ts` — pure path generator. Testable in isolation.
- `src/components/globe/use-camera-arc.ts` — hook that drives `camera.position` over time via rAF, locks `controls.enabled` during the tween, restores after.

Tween library choice: deferred to this slice's implementation start. Hand-rolled rAF + cubic-bezier ease-spring approximation is preferred (no extra dependency, ~30 lines) unless a concrete edge case justifies `gsap` or `@react-spring/three`. The interface of `use-camera-arc` is library-agnostic, so the choice can be revisited without touching `camera-arc.ts` math.

## Acceptance criteria

- [x] Click a Tier 2 pin → camera arcs to that country over 900ms with a visible altitude rise at the midpoint
- [x] During the arc, drag is disabled and additional pin clicks are ignored
- [x] Chart sheet opens to peek in parallel with the camera arc (sheet is visible during the arc, not only after)
- [x] URL `?cc=` updates immediately on click (visible in the browser address bar mid-arc, before the camera has finished moving)
- [x] Initial page load: camera starts at the default view and arcs to the picked country in 900ms; sheet animates in to peek in parallel
- [x] Opening `/?cc=br` directly: globe lands on Brazil via the same arc from the default view (shared-URL arrival experience matches in-app selection)
- [x] Unit tests: `camera-arc.ts` — at t=0 returns `from`; at t=1 returns target on sphere at baseRadius; at t=0.5 magnitude is `baseRadius * (1 + riseFactor)` within float epsilon; great-circle property: arc midpoint lies on the plane spanned by `from` and `to`
- [x] Manual verify on desktop (iOS Safari + Android Chrome deferred to Phase 4 close; antipodal cut-through math-verified — Korea↔Argentina ≈ 174°, well above SLERP fallback threshold, and antipodal regression test covers exact 180° case)

## Blocked by

#42 (Slice 1 ships the globe skeleton + pin-click → URL plumbing that this arc layers on top of)

### PR DESCRIPTION
**Pull request by taewanu**
## Summary

- Pin click and `/?cc=xx` initial entry now fly the camera in a 900ms great-circle arc with a midpoint altitude rise (`riseFactor` 0.15, `ease-spring` curve from `--ease-spring` CSS token via `bezier-easing`).
- Drag (`OrbitControls`) and additional pin clicks are locked during the arc; auto-rotate stops as soon as a country is selected.
- Tier 1 (unsupported countries) outline opacity bumped 0.5 → 0.8 — was near-invisible against the navy globe (caught during manual verify).

## Test plan

Automated (CI):
- [x] `cameraArcPath` unit tests — 4 invariants (starts at `from`, ends at target lat/lon at same distance, midpoint rises to `baseRadius × (1 + riseFactor)`, midpoint lies on great circle)
- [x] `pnpm typecheck` / `pnpm lint` / `pnpm build`

Manual (desktop Chrome, dev server):
- [x] Initial entry `/` — auto-picked country arcs from default view in 900ms
- [x] Pin click — arc to clicked country, URL `?cc=` updates immediately (before arrival)
- [x] During arc — drag locked, further pin clicks ignored, `OrbitControls` re-syncs on completion (no snap-back)
- [x] Direct URL `/?cc=br` — lands on Brazil via same arc from default view
- [x] Antipodal math — Korea/Argentina dot ≈ -0.995, well above SLERP degeneracy threshold; `riseFactor` keeps midpoint at `baseRadius × 1.15` (no cut-through possible)

Deferred:
- [ ] iOS Safari / Android Chrome manual verify — moved to Phase 4 close
- [ ] Visual antipodal verify via in-app click — blocked by far-side pin occlusion (Slice 3 hover/visibility territory)

Closes #43

## Summary by CodeRabbit

* **New Features**
  * Smooth camera animation when navigating to countries on the globe.

* **Improvements**
  * Increased visibility of country outlines for clearer map rendering.
  * Refined animation easing for more natural camera motion.

* **Tests**
  * Added geometry tests validating camera-arc behavior and edge cases.

* **Chores**
  * Added a dependency to support the new easing implementation.

[Review Change Stack](https://app.coderabbit.ai/change-stack/taewanu/sounds-abroad/pull/46?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/lib/camera-arc.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  };
+}
+
+function slerpUnitVectors(from: Vector3, to: Vector3, t: number): Vector3 {
+  const dot = Math.max(-1, Math.min(1, from.dot(to)));
+  const theta = Math.acos(dot);
+  const sinTheta = Math.sin(theta);
+  if (Math.abs(sinTheta) < 1e-6) {
+    return from.clone().lerp(to, t);
+  }
```
[the remark this answers is not in the document]
Comment (taewanu) [reply src/lib/camera-arc.ts]: Applied in 97f434d. Antipodal vectors (dot < -1 + 1e-6) now route through a perpendicular axis-angle rotation by π·t — midpoint stays on the unit sphere instead of collapsing to the origin. Near-coincident case (dot > 1 - 1e-6) takes the lerp path with explicit normalize. Regular SLERP output normalized at the end to absorb residual drift.

File: src/lib/camera-arc.test.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+test("cameraArcPath midpoint lies on the great circle from start to target", () => {
+  const fromDir = cameraStart.clone().normalize();
+  const toDir = latLonToVec3(targetLatLon.lat, targetLatLon.lon, 1);
+  const arcAxis = new Vector3().crossVectors(fromDir, toDir).normalize();
+
+  const mid = path(0.5);
+
+  expect(Math.abs(mid.dot(arcAxis))).toBeLessThan(EPSILON);
+});
```
[the remark this answers is not in the document]
Comment (taewanu) [reply src/lib/camera-arc.test.ts:50]: Added in 97f434d — `cameraArcPath survives antipodal target with non-degenerate midpoint`. Uses `{lat: 0, lon: 180}` (exact antipode of `cameraStart = (0, 0, 3.5)`). Asserts finite magnitude and the same rise invariant (`baseRadius × (1 + riseFactor)`).

File: src/components/globe/use-camera-arc.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  const [store] = useState(() => new ArcStore());
+
+  const isAnimating = useSyncExternalStore(
+    store.subscribe,
+    store.getSnapshot,
+    store.getSnapshot,
+  );
+
+  useEffect(() => {
+    if (!targetCode) return;
```
[the remark this answers is not in the document]
Comment (taewanu) [reply src/components/globe/use-camera-arc.ts:60]: Applied in 97f434d with a separate sync `useEffect` rather than direct ref assignment during render (avoids `react-hooks/refs` lint). `controls` removed from the arc effect's deps; `controlsRef.current` read at completion.
